core/src/main/java/io/questdb/griffin/engine/functions/GroupByFunction.java

```diff
@@ -74,6 +74,28 @@ default boolean earlyExit(MapValue mapValue) {
 
     int getValueIndex();
 
+    /**
+     * Called for group by function cloned to be used in different threads of parallel execution.
+     * Guaranteed to be called before any other call accessing the map or map values.
+     * {@link #initValueTypes(ArrayColumnTypes, boolean)} is not called on such functions.
+     *
+     * @param valueIndex         index of the first value of the original function in the type array
+     * @param directStrSupported flag indicating that the record used by the factory supports direct strings,
```

Comment:
is this for *all* strings? cannot it be some columns support it while others don't? (that's a question I would ask if I knew nothing about the impl and read the contract)